File: javascript/easy/relativeRanks.js
/**
 * @param {number[]} nums
 * @return {string[]}
 */
var findRelativeRanks = function(nums) {
    let res = new Array(), sorted = [...nums].sort(function(x,y){return y-x;}), ranksMap = new Map(); 
    for(let i = 0; i < sorted.length; i++) {
        if(i < 3) {
            const rank = ((i === 0) ? "Gold Medal" : ((i === 1) ? "Silver Medal" : "Bronze Medal"));
            ranksMap.set(sorted[i], rank);
        } else {
            ranksMap.set(sorted[i], (i+1).toString());
        }
    }
    for(let num of nums) {
        res.push(ranksMap.get(num));
    }
    
    return res;
};

File: ruby/easy/relative_ranks.rb
# @param {Integer[]} nums
# @return {String[]}
def find_relative_ranks(nums)
    ranks_map = Hash.new
    res = Array.new
    sorted = nums.sort.reverse
    sorted.each_with_index do |num, i|
        if i == 0
            ranks_map[num] = "Gold Medal"
        elsif i == 1
            ranks_map[num] = "Silver Medal"
        elsif i == 2
            ranks_map[num] = "Bronze Medal"
        else
            ranks_map[num] = (i+1).to_s
        end
    end

    nums.each do |num|
        res.push(ranks_map[num])
    end
    return res
end
